04_boolean/main.go

```diff
@@ -104,9 +126,34 @@ func main() {
 	// If it is not, print "You are someone else".
 	// Change the value of `firstName` and `lastName` to different values and
 	// run the program again to show all cases.
+	var firstName string = "John"
+	var lastName string = "Doe"
+	if firstName == "John" && lastName == "Doe" {
+		fmt.Println("You are John Doe")
+	} else if firstName == "John" && lastName != "Doe" {
+		fmt.Println("You are another John")
+	} else if firstName != "John" && lastName == "Doe" {
+		fmt.Println("You are another Doe")
+	} else {
+		fmt.Println("You are someone else")
+	}
 
 	// Hint: there are 2 ways to do the previous exercise:
 	// 1. Use a single level of if-else statements
 	// 2. Use nested if-else statements
 	// TODO: Try both ways.
+	if firstName == "John" {
+		if lastName == "Doe" {
+			fmt.Println("You are John Doe")
+		} else {
+			fmt.Println("You are another John")
+		}
+	}
+	if firstName != "John" {
```

Comment:
You should here use `else`.
Since the `if` before that is checking the opposite condition.

E.g.

```
if A { ... }
if !A { ... }

should be

if A { ... }
else { ... }
```

Reply:
Took me a while, but makes sense!